### PR TITLE
Remove demo sources

### PR DESCRIPTION
**Pull request by jerith**
NOTE: This PR leaves some messes that need to be cleaned up later (but not too much later!):
 * The supervision tree needs to be properly thought through and rearranged.
 * The tests need a whole lot of cleanup.

### CODE REVIEW

File: test/relay/supervisor_test.exs [hunk 1]
```diff
@@ -14,18 +16,164 @@ defmodule Relay.SupervisorTest do
...
+    "env" => %{
+      "MESOS_MARATHON_HOST" => "http://master.mesos:8080",
+      "DEBUG" => "False",
+      "PROJECT_ROOT" => "/deploy/"
+    },
+    "executor" => "",
+    "instances" => 1,
+    "labels" => %{
+      #      "MARATHON_ACME_0_DOMAIN" => "mc2.example.org",
+      #      "HAPROXY_0_REDIRECT_TO_HTTPS" => "true",
```
Comment (JayH5) [inline test/relay/supervisor_test.exs]: This stuff? Was it me who did this first and you're copy/pasting? 🤔

File: test/relay_test.exs [hunk 2]
```diff
@@ -99,35 +251,41 @@ defmodule RelayTest do
...
+    assert cds_v_3 > cds_v_2
     t1 = Time.utc_now()
     assert Time.diff(t1, t0, :milliseconds) < 1_000
 
-    # Scheduled update
-    assert_cds_request_response(streams, res_enums, "4", demo_clusters())
-    assert_lds_request_response(streams, res_enums, "3", demo_listeners())
+    # No scheduled marathon updates yet
+    # cds_v_4 = assert_cds_request_response(streams, res_enums, expected_clusters())
+    # assert cds_v_4 > cds_v_3
```
Comment (JayH5) [inline test/relay_test.exs]: Commented out?

File: test/relay/supervisor_test.exs (hunk 1 above)
Comment (jerith) [reply test/relay/supervisor_test.exs]: I'm pretty sure that's a straight copy/paste from the Marathon tests. It's probably worth trimming that down to something more or less minimal for these tests, though.

File: test/relay_test.exs (hunk 2 above)
Comment (jerith) [reply test/relay_test.exs]: We probably want that back in when the Marathon source has scheduled syncs.

File: test/relay/supervisor_test.exs (hunk 1 above)
Comment (JayH5) [reply test/relay/supervisor_test.exs]: 🙃

File: test/relay_test.exs (hunk 2 above)
Comment (JayH5) [reply test/relay_test.exs]: I would put a FIXME or something (although it's not like we go through those ever).

**Pull request merged**